http/src/main/java/io/micronaut/http/body/stream/AvailableByteArrayBody.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.body.stream;
+
+import io.micronaut.core.annotation.Experimental;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.execution.ExecutionFlow;
+import io.micronaut.core.io.buffer.ByteBuffer;
+import io.micronaut.core.io.buffer.ByteBufferFactory;
+import io.micronaut.core.util.ArgumentUtils;
+import io.micronaut.http.body.CloseableAvailableByteBody;
+import io.micronaut.http.body.InternalByteBody;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+
+/**
+ * {@link io.micronaut.http.body.AvailableByteBody} implementation based on a simple byte array.
+ *
+ * @author Jonas Konrad
+ * @since 4.6.0
+ */
+@Experimental
```

Comment:
```suggestion
@Internal
@Experimental
```

Reply:
Why did you not annotate with `@Internal`? is it because you are going to delete these classes in servlet and use this class form core in Servlet?

Reply:
correct

Reply:
I think it's fine to have this class as API now. ByteBody is already public, and the only new thing exposed here is the static factory methods